aspnetcore/security/authentication/jwt-authn.md

```diff
@@ -0,0 +1,155 @@
+| --valid-for | The period the JWT should expire after. Specify using a number followed by duration type like 'd' for days, 'h' for hours, 'm' for minutes, and 's' for seconds, e.g. 365d'. Do not use this option in conjunction with the --expires-on option. |
+| -o \| --output | The format to use for displaying output from the command. Can be one of 'default', 'token', or 'json'. |
+| -h \| --help | Show help information |
+
+## Examples
+
+Run the following commands to create an empty web project and add the [Microsoft.AspNetCore.Authentication.JwtBearer](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.JwtBearer) NuGet package:
+
+```dotnetcli
+dotnet new web -o MyJWT
+cd MyJWT
+dotnet add package Microsoft.AspNetCore.Authentication.JwtBearer --prerelease
+```
+
+Replace the contents of `Program.cs` with the following code:
+
+:::code language="csharp" source="~/security/authentication/jwt-authn/samples/MyJWT/Program.cs" id="snippet_1":::
+
+In the preceding code, a GET request to `/secret` returns an `401 Unauthorized` error. A production app might get the JWT from a [Security token service](/azure/active-directory/develop/security-tokens) (STS), perhaps in response to logging in via a set of credentials. For the purpose of working with the API during local development, the `dotnet user-jwts` command line tool can be used to create and manage app-specific local JWTs.
+
+The `user-jwts` tool is similar in concept to the  [user-secrets](xref:security/app-secrets) tool, it can be used to manage values for the app that are only valid for the developer on the local machine. In fact, the user-jwts tool utilizes the `user-secrets` infrastructure to manage the key that the JWTs are signed with, ensuring it’s stored safely in the user profile.
+
+The `user-jwts` tool hides implementation details, such as where and how the values are stored. The tool can be used without knowing the implementation details. The values are stored in a JSON file in the local machine's user profile folder:
+
+# [Windows](#tab/windows)
+
+File system path:
+
+`%APPDATA%\Microsoft\UserSecrets\<secrets_GUID>\secrets.json`
+
+# [Linux / macOS](#tab/linux+macos)
+
+File system path:
+
+`~/.microsoft/usersecrets/<secrets_GUID>/secrets.json`
+
+---
+
+### Create a JWT
+
+The following commands create a local JWT:
+
+```dotnetcli
+dotnet user-secrets init
+dotnet user-jwts create
+```
+
+In a future version, `dotnet user-jwts create` will call `dotnet user-secrets init`.
+
+The preceding command creates a JWT and updates the project’s `appsettings.Development.json` file with the JSON similar to the following:
+
+:::code language="csharp" source="~/security/authentication/jwt-authn/samples/MyJWT/appsettings.Development.json" highlight="8-21":::
+
+Copy the JWT and the `ID` created in the preceding command. The following command displays  JWT security information, including expiration, scopes, roles, token header and payload, and the compact token:
```

Comment:
As of preview7, the token gets printed when you create it so you don't have to call the print command to get it out. We probably want to document that flow.